server/src/main/java/io/crate/auth/HostBasedAuthentication.java

```diff
@@ -198,11 +211,17 @@ static boolean isValidProtocol(String hbaProtocol, Protocol protocol) {
         }
 
         static boolean isValidConnection(String hbaConnectionMode, ConnectionProperties connectionProperties) {
-            return hbaConnectionMode == null ||
-                   hbaConnectionMode.isEmpty() ||
-                   hbaConnectionMode.equals(SSL.OPTIONAL.VALUE) ||
-                   (hbaConnectionMode.equals(SSL.NEVER.VALUE) && !connectionProperties.hasSSL()) ||
-                   (hbaConnectionMode.equals(SSL.REQUIRED.VALUE) && connectionProperties.hasSSL());
+            if (hbaConnectionMode == null || hbaConnectionMode.isEmpty()) {
+                return true;
+            }
+            SSL sslMode = SSL.parseValue(hbaConnectionMode);
+            switch (sslMode) {
+                case OPTIONAL: return true;
+                case NEVER: return !connectionProperties.hasSSL();
+                case REQUIRED: return connectionProperties.hasSSL();
+                // Valid enum but unexpected case
+                default: throw new IllegalStateException(hbaConnectionMode + " is not expected HBA SSL mode");
```

Comment:
You could also use the switch expression mode here - then the compiler ensures that all cases of the enum are handled and the `default` case is no longer required.

Reply:
> including a changes entry
Yep,  it's actually related to the released code, good point. I added an entry.